File: Data_Structures/Linked_lists/insertion.c
// Write a program that inserts a new node at the beginning of a singly linked list

#include <stdio.h>
#include <stdlib.h>

struct Node {
    int data;
    struct Node* next;
};

void insertAtBeginning(struct Node** head, int newData) {
    struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
    newNode->data = newData;
    newNode->next = *head;
    *head = newNode;
}

void displayLinkedList(struct Node* head) {
    struct Node* current = head;
    while (current != NULL) {
        printf("%d ", current->data);
        current = current->next;
    }
}

int main() {
    struct Node* head = NULL;
    insertAtBeginning(&head, 3);
    insertAtBeginning(&head, 2);
    insertAtBeginning(&head, 1);
    
    printf("Linked List after insertion: ");
    displayLinkedList(head);
    printf("\n");
    
    return 0;
}

File: Data_Structures/Linked_lists/insertion.py
# Write a program that inserts a new node at the beginning of a singly linked list

class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

def insert_at_beginning(head, newData):
    newNode = Node(newData)
    newNode.next = head
    return newNode

def display_linked_list(head):
    current = head
    while current is not None:
        print(current.data, end=" ")
        current = current.next

head = None
head = insert_at_beginning(head, 1)
head = insert_at_beginning(head, 2)
head = insert_at_beginning(head, 3)

print("Linked List after insertion:", end=" ")
display_linked_list(head)
print("\n")